### DISCUSSION TITLE
CONNECTOR_RESTRICTED_USAGE

### DISCUSSION THREAD

**Discussion by HariCazorla** [International-Data-Spaces-Association/DataspaceConnector#74] [Q&A] [answered]
I am exploring the Usage control policies with the latest versions of DSC.
So far I have setup 2 local DSCs and tested the N time usage policy.

For the CONNECTOR_RESTRICTED_USAGE, I am not sure what to give for the rightOperand.
I tried with connector id from self description, connector end point and some other combinations and it is not woking as expected.

I checked the ontology file and the idsc:SYSTEM points to target system URI, I am not sure what to give?
Could you please give some inputs? 
```
{
                            "@type": "ids:Permission",
                            "@id": "https://w3id.org/idsa/autogen/permission/75b7fe03-8f89-42f7-8ef4-bd43b14642be",
                            "ids:description": [
                                {
                                "@value": "connector-restriction",
                                "@type": "http://www.w3.org/2001/XMLSchema#string"
                                }
                            ],
                            "ids:preDuty": null,
                            "ids:postDuty": null,
                            "ids:constraint": [
                                {
                                "@type": "ids:Constraint",
                                "@id": "https://w3id.org/idsa/autogen/constraint/4166415b-5509-4779-8d8a-9821fcea572e",
                                "ids:leftOperand": {
                                    "properties": null,
                                    "@id": "idsc:SYSTEM"
                                },
                                "ids:operator": {
                                    "properties": null,
                                    "@id": "idsc:SAME_AS"
                                },
                                "ids:rightOperand": {
                                    "@value": "https://connector1:8080/api/ids/data",
                                    "@type": "xsd:anyURI"
                                },
                                "ids:rightOperandReference": null,
                                "ids:unit": null,
                                "ids:pipEndpoint": null
                                }
                            ],
                            "ids:action": [
                                {
                                "properties": null,
                                "@id": "idsc:USE"
                                }
                            ],
                            "ids:assetRefinement": null,
                            "ids:title": [
                                {
                                "@value": "Example Usage Policy",
                                "@type": "http://www.w3.org/2001/XMLSchema#string"
                                }
                            ],
                            "ids:assignee": null,
                            "ids:assigner": null,
                            "ids:target": null
            }
```

**Comment by juliapampus** [comment, marked as answer]
The `"ids:rightOperand": { "@value": "https://connector1:8080/api/ids/data", "@type": "xsd:anyURI" },` should point to the connector's id. The matching of the IDs only works, if the connector's `@id` in the `config.json` is set correctly.

**Reply by HariCazorla** [reply]
Thanks for the answer, I will check this and let you know about it.
One small clarification, could you please let me know if we can just change the UUID to get a different connector id? or is there any procedure to generate an id of the connector? 
`
"ids:connectorDescription": {
    "@type": "ids:BaseConnector",
    "@id": "https://w3id.org/idsa/autogen/baseConnector/7b934432-a85e-41c5-9f65-669219dde4ea",
`

**Reply by HariCazorla** [reply]
I tried your suggestions, it is working as expected. I changed the uuid of the connectors manullay on the config.json file.
I have 3 DSC, one provider, two consumers and only one of them can get the reource. When I fetch data from other DSC, the error message is not proper, ">‰bs*Þ²ÚârØ¨×­yË^tú%‰Ì«zËk‰Ëb¢w^µç-y". Is this an expected behaviour?
